File: 00-Text-Field/00-Text-Field/ViewController.m
//
//  ViewController.m
//  00-Text-Field
//
//  Created by meloMacPro on 2018/9/12.
//  Copyright © 2018年 meloMacPro. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property(nonatomic, strong) UILabel *prefixLabel;
@property(nonatomic, strong) UITextField *textField;
@end

@implementation ViewController

- (UILabel *)prefixLabel {
    if (!_prefixLabel) {
        _prefixLabel = [[UILabel alloc]initWithFrame:CGRectZero];
        //This sets the label text
        _prefixLabel.text =@" ## ";
        // This sets the font for the label
        [_prefixLabel setFont:[UIFont boldSystemFontOfSize:14]];
        // This fits the frame to size of the text
        [_prefixLabel sizeToFit];
    }
    return _prefixLabel;
}

- (UITextField *)textField {
    if (!_textField) {
        _textField = [[UITextField  alloc] initWithFrame:
                      CGRectMake((self.view.frame.size.width-80)/2.0, (self.view.frame.size.height-30)/2.0, 80, 30)];
        
        // This sets the border style of the text field
        _textField.borderStyle = UITextBorderStyleRoundedRect;
        _textField.contentVerticalAlignment =
        UIControlContentVerticalAlignmentCenter;
        [_textField setFont:[UIFont boldSystemFontOfSize:12]];
        
        //Placeholder text is displayed when no text is typed
        _textField.placeholder = @"Simple Text Field";
    }
    return _textField;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view addSubview:self.textField];
    [self.view addSubview:self.prefixLabel];
    self.textField.leftView = self.prefixLabel;
    
    //It set when the left prefixLabel to be displayed
    self.textField.leftViewMode = UITextFieldViewModeAlways;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: S00-Text-Field/S00-Text-Field/ViewController.swift
//
//  ViewController.swift
//  S00-Text-Field
//
//  Created by meloair on 2018/9/12.
//  Copyright © 2018年 com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    
    lazy var label : UILabel = {
        let object = UILabel(frame: CGRect.zero)
        object.text = "@"
        object.font = UIFont.boldSystemFont(ofSize: 14)
        object.sizeToFit()
        return object
    }()
    
    lazy var textField : UITextField = {
        let object = UITextField(frame: CGRect(x:(self.view.frame.size.width - 80)/2, y:(self.view.frame.size.height - 50)/2, width:80, height:50))
        object.placeholder = "Simple Text Field"
        object.textAlignment = .center
        object.font = UIFont.boldSystemFont(ofSize: 14)
        return object
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.view.addSubview(self.label)
        
        self.textField.leftView = self.label
        self.textField.leftViewMode = UITextFieldViewMode.always
        
        self.view.addSubview(self.textField)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

